pkg/api/handlers/compat/images_build.go

```diff
@@ -391,20 +391,47 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Docker's newer clients popuates `cacheFrom` and `cacheTo` parameter
```

Comment:
"Newer" is a danger-bell word. Please rewrite this as "Docker client, as of vX.Y, ..." (or "as of late 2022" if exact version is hard to find).

Reply:
Actually here "newer" already stands for something rather old. I am not sure Docker client 18.09 had this issue, but I could confirm Docker client 19.03.0 already exposed the problem (see the value `cachefrom=%5B%5D` in what follows, showing the usage of an explicit empty array).

```
[root@771a32fa8a92 plop]# dnf install -y https://download.docker.com/linux/centos/7/x86_64/stable/Packages/docker-ce-cli-19.03.0-3.el7.x86_64.rpm
Last metadata expiration check: 0:12:08 ago on Thu Jan  5 22:24:12 2023.
docker-ce-cli-19.03.0-3.el7.x86_64.rpm                                                                                                                                                 84 MB/s |  39 MB     00:00
Package docker-ce-cli-1:19.03.0-3.el7.x86_64 is already installed. 
Dependencies resolved.
Nothing to do.
Complete!
[root@771a32fa8a92 plop]# docker build .
Sending build context to Docker daemon  2.048kB
ERRO[0000] failed to dial gRPC: unable to upgrade to h2c, received 404
error during connect: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.40/build?buildargs=%7B%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&rm=1&session=l6z7cmb92kk629s4k6vd92czr&shmsize=0&target=&ulimits=null&version=1: context canceled
[root@771a32fa8a92 plop]# docker --version 
Docker version 19.03.0, build aeac9490dc
```